ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoiceTest.java

```diff
@@ -233,6 +238,86 @@ void onBlock_shouldFailIfBlobsAreNotAvailable() {
     verify(blobSidecarsAvailabilityChecker).getAvailabilityCheckResult();
   }
 
+  @Test
+  void onBlock_consensusValidationShouldNotResolveWhenDataAvailabilityFails() {
+    setupWithSpec(TestSpecFactory.createMinimalDeneb());
+    consensusValidationResult = Optional.of(new SafeFuture<>());
+    final SignedBlockAndState blockAndState = chainBuilder.generateBlockAtSlot(ONE);
+    storageSystem.chainUpdater().advanceCurrentSlotToAtLeast(blockAndState.getSlot());
+
+    when(blobSidecarsAvailabilityChecker.getAvailabilityCheckResult())
+        .thenReturn(SafeFuture.completedFuture(BlobSidecarsAndValidationResult.NOT_AVAILABLE));
+
+    importBlockWithError(blockAndState, FailureReason.FAILED_DATA_AVAILABILITY_CHECK_NOT_AVAILABLE);
+
+    assertThatSafeFuture(consensusValidationResult.get()).isNotDone();
+
+    verify(blobSidecarManager).createAvailabilityChecker(blockAndState.getBlock());
+    verify(blobSidecarsAvailabilityChecker).initiateDataAvailabilityCheck();
+    verify(blobSidecarsAvailabilityChecker).getAvailabilityCheckResult();
+  }
+
+  @Test
+  void onBlock_consensusValidationShouldNotResolveWhenEarlyFails() {
+    setupWithSpec(TestSpecFactory.createMinimalDeneb());
+    consensusValidationResult = Optional.of(new SafeFuture<>());
+    final List<SignedBlockAndState> signedBlockAndStates = chainBuilder.generateBlocksUpToSlot(2);
+    final SignedBlockAndState wrongBlockAndState =
+        signedBlockAndStates.get(signedBlockAndStates.size() - 1);
+
+    storageSystem.chainUpdater().advanceCurrentSlotToAtLeast(wrongBlockAndState.getSlot());
+
+    importBlockWithError(wrongBlockAndState, FailureReason.UNKNOWN_PARENT);
+
+    assertThatSafeFuture(consensusValidationResult.get()).isNotDone();
+  }
+
+  @Test
+  void onBlock_consensusValidationShouldNotResolveWhenStateTransitionFails()
+      throws StateTransitionException {
+    setupWithSpec(TestSpecFactory.createMinimalDeneb());
+    consensusValidationResult = Optional.of(new SafeFuture<>());
+    final SignedBlockAndState blockAndState = chainBuilder.generateBlockAtSlot(ONE);
+    storageSystem.chainUpdater().advanceCurrentSlotToAtLeast(blockAndState.getSlot());
+
+    final BlockProcessor blockProcessor = mock(BlockProcessor.class);
+    when(spec.getBlockProcessor(blockAndState.getSlot())).thenReturn(blockProcessor);
+    when(blockProcessor.processAndValidateBlock(any(), any(), any(), any()))
+        .thenThrow(new StateTransitionException("error!"));
+
+    importBlockWithError(blockAndState, FailureReason.FAILED_STATE_TRANSITION);
```

Comment:
Just noticed the name is a bit confusing. Not a scope of this PR but maybe something like `assertBlockImportedWithError`